src/commands/protocols/common/submit-commit-command.js

```diff
@@ -48,6 +48,32 @@ class SubmitCommitCommand extends Command {
             );
         }
 
+        const assertionIds = await this.blockchainModuleManager.getAssertionIds(
+            blockchain,
+            contract,
+            tokenId,
+        );
+
+        // If update for new state is already finalized (and node haven't processed the event yet), don't send commit for the older state
+        if (stateIndex < assertionIds.length - 1) {
+            this.logger.trace(
+                `Knowledge Asset was updated, not sending Commit for the Service Agreement with the ID: ${agreementId}, ` +
+                    `Blockchain: ${blockchain}, Contract: ${contract}, Token ID: ${tokenId}, ` +
+                    `Keyword: ${keyword}, Hash function ID: ${hashFunctionId}, Epoch: ${epoch}, ` +
+                    `State Index: ${stateIndex}, Operation ID: ${operationId}`,
+            );
+
+            this.operationIdService.emitChangeEvent(
```

Comment:
If we emit this event, it will be calculated as successful update commit in telemetry and i guess that we don't want that. We should probably remove this emit?

Reply:
Hm, we do the same for "Already Submitted", I thought we should always end command with some status, aren't we?

Reply:
Yes we have the same thing on other places, but i'm thinking now maybe we were wrong. We don't want to count this commit so we don't need end event. If there is no end event it will just be discarder during signal processing.

Reply:
Right now we don't want to count commits/proofs that are intentionally stopped by the node. We want to have count of successful ones and the one that failed with errors that are not expected. If we are handling some case where we detect that node does not want to commit, then at the moment it is easier to not close the event, so we don't count it in telemetry

Reply:
@djordjekovac @NZT48 then I think it makes sense to remove this end event as well for other cases, what do you think?